bundles/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVStationHandler.java

```diff
@@ -0,0 +1,155 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * KVVStationHandler represents a station and holds information about the trains
+ * which will arrive soon.
+ *
+ * @author Maximilian Hess - Initial contribution
+ */
+@NonNullByDefault
+public class KVVStationHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(KVVStationHandler.class);
+
+    private final KVVStationConfig config;
+
+    public KVVStationHandler(final Thing thing) {
+        super(thing);
+        logger.info("stationhandler!");
+        this.config = getConfigAs(KVVStationConfig.class);
+    }
+
+    @Override
+    public void initialize() {
+        scheduler.execute(() -> {
+
+            // creating channels
+            logger.info("Creating channels...");
+            final List<Channel> channels = new ArrayList<Channel>();
+            for (int i = 0; i < this.config.maxTrains; i++) {
+                channels.add(ChannelBuilder.create(
+                    new ChannelUID(this.thing.getUID(), "train" + i + "-name"), "String").build());
+                channels.add(ChannelBuilder.create(
+                    new ChannelUID(this.thing.getUID(), "train" + i + "-destination"), "String").build());
+                channels.add(ChannelBuilder.create(
+                    new ChannelUID(this.thing.getUID(), "train" + i + "-eta"), "String").build());
+            }
+            this.updateThing(this.editThing().withChannels(channels).build());
+
+            logger.info("Starting inital fetch");
+            final Bridge bridge = getBridge();
+            if (bridge == null) {
+                logger.warn("Failed to get bridge (is null)");
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+
+            final KVVBridgeHandler handler = (KVVBridgeHandler)  bridge.getHandler();
+            if (handler == null) {
+                logger.warn("Failed to get bridge handler (is null)");
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+
+            final DepartureResult departures = handler.queryKVV(this.config);
+            if (departures == null) {
+                logger.warn("Failed to get departures for '{}'", this.thing.getUID().getAsString());
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            }
+
+            logger.info("Listing channels...");
+            for (final Channel c : this.getThing().getChannels()) {
+                logger.info("{}", c.getUID().getAsString());
+            }
+            
+            this.setDepartures(departures);
+            this.scheduler.scheduleWithFixedDelay(new UpdateTask(), 0, this.config.updateInterval,
```

Comment:
You should store the `ScheduledFuture` returned by `scheduleWithFixedDelay()` and cancel the timer in the Handler's `dispose()` method. See https://www.openhab.org/docs/developer/bindings/#polling-for-a-state

Reply:
Solved.